src/readers/morphologyHDF5.cpp

```diff
@@ -70,193 +87,154 @@ Property::Properties load(const HighFive::Group& group) {
+    } else if (pointsDims[1] != _pointColumns) {
+        throw RawDataError("Opening morphology '" + _uri +
+                           "': incorrect number of columns for points");
     }
-}
 
+    std::vector<std::array<floatType, _pointColumns>> hd5fData(numberPoints);
 
-static inline bool HasNeurites(int firstSectionOffset) {
-    return (firstSectionOffset > -1);
-}
+    if (hd5fData.size() > 0) {
+        pointsDataSet.read(hd5fData.front().data());
+    }
 
-void MorphologyHDF5::_readPoints(int firstSectionOffset) {
-    auto& points = _properties.get<Property::Point>();
-    auto& diameters = _properties.get<Property::Diameter>();
+    const bool hasNeurites = size_t(firstSectionOffset) < numberPoints;
+    const size_t section_offset = hasNeurites ? size_t(firstSectionOffset) : hd5fData.size();
 
     auto& somaPoints = _properties._somaLevel._points;
     auto& somaDiameters = _properties._somaLevel._diameters;
 
-    auto loadPoints =
-        [&](const std::vector<std::array<morphio::floatType, _pointColumns>>& hd5fData,
-            bool hasNeurites) {
-            const std::size_t section_offset = hasNeurites ? std::size_t(firstSectionOffset)
-                                                           : hd5fData.size();
-
-            // points and diameters are PODs. Fastest to resize then assign values
-            somaPoints.resize(somaPoints.size() + section_offset);
-            somaDiameters.resize(somaDiameters.size() + section_offset);
-            for (std::size_t i = 0; i < section_offset; ++i) {
-                const auto& p = hd5fData[i];
-                somaPoints[i] = {p[0], p[1], p[2]};
-                somaDiameters[i] = p[3];
-            }
+    somaPoints.resize(somaPoints.size() + section_offset);
+    somaDiameters.resize(somaDiameters.size() + section_offset);
 
+    for (size_t i = 0; i < section_offset; ++i) {
+        const auto& p = hd5fData[i];
+        somaPoints[i] = {p[0], p[1], p[2]};
+        somaDiameters[i] = p[3];
+    }
 
-            if (hasNeurites) {
-                const size_t size = (points.size() + hd5fData.size() - section_offset);
-                points.resize(size);
-                diameters.resize(size);
-                for (std::size_t i = section_offset; i < hd5fData.size(); ++i) {
-                    const auto& p = hd5fData[i];
-                    const std::size_t section_i = i - section_offset;
-                    points[section_i] = {p[0], p[1], p[2]};
-                    diameters[section_i] = p[3];
-                }
-            }
-        };
+    auto& points = _properties.get<Property::Point>();
+    auto& diameters = _properties.get<Property::Diameter>();
 
-    std::vector<std::array<morphio::floatType, _pointColumns>> vec(_pointsDims[0]);
-    if (vec.size() > 0) {
-        _points->read(vec.front().data());
+    if (hasNeurites) {
+        const size_t size = (points.size() + hd5fData.size() - section_offset);
```

Comment:
same for points.size() it should/must be 0 no ?

Reply:
fixed